src/Planner/PlannerJoinTree.cpp

```diff
@@ -143,6 +151,106 @@ NameAndTypePair chooseSmallestColumnToReadFromStorage(const StoragePtr & storage
+        settings.allow_experimental_query_deduplication || //
+        settings.empty_result_for_aggregation_by_empty_set)
+        return false;
+
+    QueryTreeNodes aggregates = collectAggregateFunctionNodes(query_tree);
+    if (aggregates.size() != 1)
+        return false;
+
+    const auto * function_node = typeid_cast<const FunctionNode *>(aggregates.front().get());
+    if (!function_node)
+        return false;
+
+    if (!function_node->getAggregateFunction())
+        return false;
+
+    const auto * count_func = typeid_cast<const AggregateFunctionCount *>(function_node->getAggregateFunction().get());
+    if (!count_func)
+        return false;
+
+    /// get number of rows
+    std::optional<UInt64> num_rows{};
+    /// Transaction check here is necessary because
+    /// MergeTree maintains total count for all parts in Active state and it simply returns that number for trivial select count() from table query.
+    /// But if we have current transaction, then we should return number of rows in current snapshot (that may include parts in Outdated state),
+    /// so we have to use totalRowsByPartitionPredicate() instead of totalRows even for trivial query
+    /// See https://github.com/ClickHouse/ClickHouse/pull/24258/files#r828182031
+    if (!main_query_node.hasPrewhere() && !main_query_node.hasWhere() && !query_context->getCurrentTransaction())
+    {
+        num_rows = storage->totalRows(settings);
+    }
+    // TODO:
+    // else // It's possible to optimize count() given only partition predicates
+    // {
+    //     SelectQueryInfo temp_query_info;
+    //     temp_query_info.query = query_ptr;
+    //     temp_query_info.syntax_analyzer_result = syntax_analyzer_result;
+    //     temp_query_info.prepared_sets = query_analyzer->getPreparedSets();
+    //     num_rows = storage->totalRowsByPartitionPredicate(temp_query_info, context);
+    // }
+
+    if (!num_rows)
+        return false;
+
+    /// set aggregation state
+    const AggregateFunctionCount & agg_count = *count_func;
+    std::vector<char> state(agg_count.sizeOfData());
+    AggregateDataPtr place = state.data();
+    agg_count.create(place);
+    SCOPE_EXIT_MEMORY_SAFE(agg_count.destroy(place));
+    agg_count.set(place, num_rows.value());
+
+    auto column = ColumnAggregateFunction::create(function_node->getAggregateFunction());
+    column->insertFrom(place);
+
+    /// get count() argument type
+    DataTypes argument_types;
+    argument_types.reserve(columns_names.size());
+    {
+        const Block source_header = table_node.getStorageSnapshot()->getSampleBlockForColumns(columns_names);
+        for (const auto & column_name : columns_names)
+            argument_types.push_back(source_header.getByName(column_name).type);
+    }
+
+    Block block_with_count{
+        {std::move(column),
+         std::make_shared<DataTypeAggregateFunction>(function_node->getAggregateFunction(), argument_types, Array{}),
+         columns_names.front()}};
```

Comment:
The name of this column in block probably should be `count(column_name)` or `count()`. To be safe better to use `calculateActionNodeName` from PlannerActionsVisitor.
I expect that this should work `SELECT count() + 1 FROM test_table`.

Reply:
The implementation already returns correct columns, for example:
```
SELECT count() + 1
FROM m3
SETTINGS max_rows_to_read = 1

Query id: 488ad40d-402c-407c-b2d1-3b8633391434

┌─plus(count(), 1)─┐
│                3 │
└──────────────────┘
```

With your suggestion, the column name in projection will be `plus(count(), 1_UInt8)`